nemo/collections/asr/modules/flashlight_decoder.py

```diff
@@ -0,0 +1,253 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+import torch
+import itertools as it
+import numpy as np
+
+from typing import Iterable, List, Optional, Tuple, Union
+
+from nemo.core.classes import NeuralModule, typecheck
+from nemo.core.neural_types import LengthsType, LogprobsType, NeuralType, PredictionsType
+from nemo.collections.common.tokenizers.tokenizer_spec import TokenizerSpec
+
+
+class FlashLightKenLMBeamSearchDecoder(NeuralModule):
+    '''
+    @property
+    def input_types(self):
+        """Returns definitions of module input ports.
+        """
+        return {
+            "log_probs": NeuralType(('B', 'T', 'D'), LogprobsType()),
+        }
+
+    @property
+    def output_types(self):
+        """Returns definitions of module output ports.
+        """
+        return {"hypos": NeuralType(('B'), PredictionsType())}
+    '''
+    
+    class TokensWrapper:

```

Comment:
I made it inner because it's just a special helper class just for the flashlight_decoder and is necessary for it to work, and it has no use outside of that. I think it's better to keep it as an inner class for this reason

Reply:
Pickle has an issue with inner classes, and we may need to eventually support pickle so lets extract it out as a private class in this file.

Reply:
Pickle doesn't work for Flashlight regardless, I have already tried, with or without the inner classes, you can't pickle this module for multiprocessing for faster decoding, I already tried

Reply:
Fixed